### ISSUE TITLE
Inherited un-versioned object didn't publish assets when creating new object.

### ISSUE THREAD

**Issue by mmativ** [silverstripe/silverstripe-versioned#226]
Version: 4.3.3

I extend page type with new page type then while im making new page i notice that my assets are still in draft status but its working fine on parent page type so i add `$owns` to object and its working fine, it should publish the assets when creating new object right?

**Parent Page Type:**

```
class ParentPage extends Page
{
    private static $has_many = [
        'DataObject' => DataObject::class
    ];
}
```

**Child Page Type**

```
class ChildPage extends ParentPage
{
}
```

**Data Object**

```
class ProductImage extends DataObject
{
    private static $has_one = [
        'ParentPage' => ParentPage::class,
        'Image' => Image::class
    ];

    private static $owns = [
        'Image'
    ];
}
```

**Comment by dnsl48** [comment]
Hi @mmativ 
From the snippets you provide it doesn't look like your ParentPage actually owns the DataObject.

Here's a simple example of how it could look like:
```php
<?php
use SilverStripe\Versioned\Versioned;
use SilverStripe\Assets\Image;
use SilverStripe\ORM\DataObject;
use Page;

class MyPage extends Page
{
    private static $has_many = [
        'Banners' => Banner::class
    ];
    private static $owns = [
        'Banners'
    ];
}

class Banner extends DataObject
{
    private static $extensions = [
        Versioned::class
    ];
    private static $has_one = [
        'Parent' => MyPage::class,
        'Image' => Image::class,
    ];
    private static $owns = [
        'Image'
    ];
}
```

If you need more advice, please consider joining the [SilverStripe communtiy slack channel](https://www.silverstripe.org/community/slack-signup), [SilverStripe user forums](https://forum.silverstripe.org/) or if you believe your question might help others then you might ask it on [StackOverflow](https://stackoverflow.com/questions/tagged/silverstripe) tagging it with `silverstripe`.

This particular issue does not look like a bug in the module, so I'm closing it.
However, if you believe you have found a bug, please feel free to reopen.

**Comment by mmativ** [comment]
> Hi @mmativ
> From the snippets you provide it doesn't look like your ParentPage actually owns the DataObject.
> 
> Here's a simple example of how it could look like:
> 
> ```
> <?php
> use SilverStripe\Versioned\Versioned;
> use SilverStripe\Assets\Image;
> use SilverStripe\ORM\DataObject;
> use Page;
> 
> class MyPage extends Page
> {
>     private static $has_many = [
>         'Banners' => Banner::class
>     ];
>     private static $owns = [
>         'Banners'
>     ];
> }
> 
> class Banner extends DataObject
> {
>     private static $extensions = [
>         Versioned::class
>     ];
>     private static $has_one = [
>         'Parent' => MyPage::class,
>         'Image' => Image::class,
>     ];
>     private static $owns = [
>         'Image'
>     ];
> }
> ```
> 
> If you need more advice, please consider joining the [SilverStripe communtiy slack channel](https://www.silverstripe.org/community/slack-signup), [SilverStripe user forums](https://forum.silverstripe.org/) or if you believe your question might help others then you might ask it on [StackOverflow](https://stackoverflow.com/questions/tagged/silverstripe) tagging it with `silverstripe`.
> 
> This particular issue does not look like a bug in the module, so I'm closing it.
> However, if you believe you have found a bug, please feel free to reopen.

Yeah that's right but im not using versioned data object here then just to inform you we already discuss this on slack so i just make an issue so someone can fix it.

**Comment by mmativ** [comment]
@dnsl48 can we re open this one?

cc: @kinglozzer

**Comment by dnsl48** [comment]
@mmativ Are you saying the snippet I provided doesn't work if I remove Versioned extension from there? Or is it something else?

**Comment by mmativ** [comment]
@dnsl48 It will work but if you extend the `MyPage` and add `Banner` items, image under banner will not be published.

**Comment by kinglozzer** [comment]
From my memory of the Slack conversation, the objective was to have the `Banner` DataObject _not_ versioned, but still use `$owns` to ensure that the image attached to each banner is published as soon as the DataObject is saved.

@mmativ reported that it worked on `ParentPage`, but when using `ChildPage` the related images were not published when the DataObjects were saved, despite inheriting the same `GridField` etc for managing them.

**Comment by dnsl48** [comment]
Ok. @mmativ could you please provide the code snippets to clarify the bug? Otherwise, it's not clear what actual problem is in this issue.

**Comment by ScopeyNZ** [comment]
Whoops, wrong button - sorry for the noise.
